### PR TITLE
Video block v2

### PR DESCRIPTION
**Pull request by sophieH29**
fv1
fv2
fv5
fv3_m
fv4_m

### CODE REVIEW

File: scripts/redux/constants.js [hunk 1]
```diff
@@ -7,3 +7,7 @@ const SET_HERO = 'SET_HERO';
 const FETCH_TICKETS = 'FETCH_TICKETS';
 // partners
 const FETCH_PARTNERS = 'FETCH_PARTNERS';
+// featured videos
+const FETCH_VIDEOS = 'FETCH_VIDEOS';
+// video dialog
+const TOGGLE_DIALOG = 'TOGGLE_DIALOG';
```
Comment (ozasadnyy) [inline scripts/redux/constants.js]: Can we use this constant to invoke other dialogs, like speaker's, session's, etc. ?

File: src/elements/featured-videos.html [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        height: 250px;
+        overflow: hidden;
+        margin-left: -15px;
+      }
+
+      .videos {
+        transition: all 0.4s cubic-bezier(0, 0, 0.2, 1);
+        will-change: transition;
+        backface-visibility: hidden;
+        transform: translateX(0px);
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: remove `px`

File: src/elements/featured-videos.html [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        position: relative;
+        margin: 0 auto;
+        max-width: 1280px;
+        height: 250px;
+        overflow: hidden;
+        margin-left: -15px;
+      }
+
+      .videos {
+        transition: all 0.4s cubic-bezier(0, 0, 0.2, 1);
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: let's define and use one animation everywhere. Right now, we have defined animation in `shared-styles` https://github.com/gdg-x/hoverboard/blob/hoverboard-v2/src/elements/shared-styles.html#L27

We can discuss which animation will work for us in the best way

File: src/elements/featured-videos.html [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+      .preview-image {
+        display: block;
+        cursor: pointer;
+        width: 280px;
+        min-width: 270px;
+        height: 177px;
+        border-radius: 5px;
+        margin-bottom: 10px;
+        overflow: hidden;
+        box-sizing: border-box;
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: Unnecessary as defined in shared-styles
https://github.com/gdg-x/hoverboard/blob/hoverboard-v2/src/elements/shared-styles.html#L32

File: src/elements/featured-videos.html [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+          height: 260px;
+        }
+
+         .video-list {
+          min-height: 300px;
+         }
+      }
+    </style>
+
+    <iron-media-query id="mq-phone" full query="(max-width: {$ mediaQueries.xs.max $})" query-matches="{{isPhoneSize}}"></iron-media-query>
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: No need to define it. You can use value from redux-store
https://github.com/gdg-x/hoverboard/blob/hoverboard-v2/src/redux-mixin.html#L14

Example: https://github.com/gdg-x/hoverboard/blob/hoverboard-v2/src/elements/tickets-block.html#L140

File: src/elements/featured-videos.html [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+          background-color: var(--default-primary-color);
+          border-radius: 50%;
+        }
+
+        .video-list:hover .chevron-icon {
+          display: block;
+        }
+
+        .next-video {
+          justify-content: flex-end;
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: You can use attributes instead

File: src/elements/featured-videos.html [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+        }
+
+        shiftContentLeft() {
+          const {
+            cardWidth,
+            videosLength
+          } = this.getVideosDetails();
+
+          if (this._currentIndex < videosLength && this._currentIndex > 0) {
+            let newX = -(this._currentIndex - 1) * cardWidth;
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: should be `const` as we don't change it

File: src/elements/featured-videos.html [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+          let videosLength = videos.length;
+
+          return {
+            cardWidth,
+            videosLength
+          }
+        }
+
+        getVideoListWidth() {
+          let videos = this.shadowRoot.querySelectorAll('.video-item');
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: `const`

File: src/elements/featured-videos.html [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+        transformVideoList(el, newPosition) {
+          el.style.transform = 'translate3d(' + newPosition + 'px, 0, 0)';
+        }
+
+        getSessionsImageUrl(path, index) {
+          var id = '#image' + index;
+          firebase.storage()
+            .ref(path)
+            .getDownloadURL()
+            .then(function (url) {
```
Comment (ozasadnyy) [inline src/elements/featured-videos.html]: use arrow functions + if you do this you will be able to remove `.bind(this)`

File: src/elements/partners-block.html [hunk 10]
```diff
@@ -5,8 +5,6 @@
 <link rel="import" href="../../bower_components/iron-icon/iron-icon.html">
 
 <link rel="import" href="../redux-mixin.html">
-<link rel="import" href="./shared-styles.html">
```
Comment (ozasadnyy) [inline src/elements/partners-block.html]: Why did you remove them?

File: src/elements/video-dialog.html [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+<link rel="import" href="../../bower_components/paper-icon-button/paper-icon-button.html">
+<link rel="import" href="../../bower_components/paper-dialog-behavior/paper-dialog-behavior.html">
+<link rel="import" href="../../bower_components/paper-button/paper-button.html">
+
+<dom-module id="video-dialog">
+    <template>
+
+        <style>
+             :host {
+                position: absolute;
```
Comment (ozasadnyy) [inline src/elements/video-dialog.html]: `fit` attribute https://github.com/gdg-x/hoverboard/blob/hoverboard-v2/src/elements/flex-layout-attr.html#L369

File: src/elements/video-dialog.html [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+                color: #fff;
+                opacity: 0.8;
+                background-color: var(--default-primary-color);
+                border-radius: 50%;
+                display: block;               
+                margin-top: 5px;
+            }
+
+            .video {
+                position: absolute;
```
Comment (ozasadnyy) [inline src/elements/video-dialog.html]: `fit`

File: src/elements/video-dialog.html [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+        class VideoDialog extends Polymer.mixinBehaviors([Polymer.PaperDialogBehavior], Polymer.Element) {
+
+            static get is() {
+                return 'video-dialog'
+            }
+
+            static get properties() {
+                return {
+                    videoTitle: {
+                        type: String
```
Comment (ozasadnyy) [inline src/elements/video-dialog.html]: Let's define code style for such case.
```js
videoTitle: String
```
or as you have.

File: src/elements/video-dialog.html [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+                    disableControls: {
+                        type: Boolean,
+                        value: false
+                    }
+                }
+            }
+
+            static get observers() {
+                return [
+                    'videoDialogActionMade(opened)'
```
Comment (ozasadnyy) [inline src/elements/video-dialog.html]: you can add observer in the `properties` section

File: scripts/redux/constants.js (hunk 1 above)
Comment (sophieH29) [inline scripts/redux/constants.js]: I think yes, that's why I put it in Redux

File: src/elements/partners-block.html (hunk 10 above)
Comment (sophieH29) [inline src/elements/partners-block.html]: shared styles file is important from parent component, so it's already have it. Do you think it's better to include this in each component?

File: src/elements/video-dialog.html (hunk 13 above)
Comment (sophieH29) [inline src/elements/video-dialog.html]: lets use yours

File: src/elements/partners-block.html (hunk 10 above)
Comment (ozasadnyy) [inline src/elements/partners-block.html]: I do. @robdodson told me that, but it was early 2015. We have to ask about this in Polymer slack channel

Comment (sophieH29) [inline src/elements/partners-block.html]: I think it's about reusable components. When the component should be independent, have everything inside it and ready to be used from any other places/projects etc.
But when we specify relative paths to other files in projects (not from bower_components folder), the component is dependent, so in that case we don't need to have such imports in every file, coz it gets that from parent
